backend/controllers/whereabouts.js

```diff
@@ -0,0 +1,95 @@
+const moment = require('moment')
+const { merge, switchDateFormat, getCurrentShift, pascalToString } = require('../utils')
+
+const getReasonCountMap = (data, reasons) => {
+  const filterAttendance = reason =>
+    data.filter(
+      attendance =>
+        attendance && attendance.absentReason && attendance.absentReason.toLowerCase() === reason.toLowerCase()
+    ).length
+
+  return Object.values(reasons)
+    .map(absentReason => ({
+      [absentReason.toLowerCase()]: filterAttendance(absentReason),
+    }))
+    .reduce(merge, {})
+}
+
+const whereaboutsDashboardFactory = (oauthApi, elite2Api, whereaboutsApi) => {
+  const getDashboardStats = async (context, { agencyId, period, date }) => {
+    const attendances = await whereaboutsApi.getPrisonAttendance(context, { agencyId, period, date })
+    const scheduledActivities = await elite2Api.getOffenderActivities(context, { agencyId, period, date })
+    const absentReasons = await whereaboutsApi.getAbsenceReasons(context)
```

Comment:
We should pass the absent reasons in from whereaboutsDashboard line 61. There's no need to call get absent reasons twice.